### ISSUE TITLE
UI of Armory Exchange hidden error

### ISSUE THREAD

**Issue by chaoticnote** [danreeves/dt-exchange#79]
![image](https://github.com/danreeves/dt-exchange/assets/28378016/790993c3-1a24-42d9-acde-f01e72554715)

Been a user of this for months, nearly since the 1st month of this extension's release. First time seeing this issue personally, where the list of the armory doesn't show up.

Below is the console log with a total of four errors.

![image](https://github.com/danreeves/dt-exchange/assets/28378016/f93b6f17-e3cf-4a9f-adaa-6062f8ab6aba)
![image](https://github.com/danreeves/dt-exchange/assets/28378016/7c1c636a-9237-40e3-ae5f-6d277f31cbd9)
![image](https://github.com/danreeves/dt-exchange/assets/28378016/5f90acb5-7dab-41da-a69f-e048c08946ef)
![image](https://github.com/danreeves/dt-exchange/assets/28378016/33a46cf7-a955-4dbb-88f7-d4495620758e)

**Comment by wrubiks** [comment]
Having the same issue, it loads one or 2 characters every couple hours.

**Comment by JihadFenix** [comment]
I've been having the exact same issue since 20231206. Watching this.

**Comment by bzly** [comment]
I believe this occurs where characters have any of the new weapons in their store list. It looks like we're trying to get the `display_name` of items that can't be found. 

We probably need to have them added to src/localisation.json

Does anyone have a good way of getting all of the new weapon ids? If one inspects the response from the `credit_store_<class>` endpoint for a failing shop-rotation, they can extract the id of the item once identified—e.g. the new revolver is `content/items/weapons/player/ranged/stubrevolver_p1_m2`. Presumably there's an easy way to do this if you're at all experienced with Darktide modding, and can rip from the game files. 

New weapons:
| display_name | id |
| --- | --- |
| Munitorum Mk III Sapper Shovel |  |
| Munitorum Mk VII Sapper Shovel | content/items/weapons/player/melee/combataxe_p3_m3 |
| Brute-Brainer Mk XIX Latrine Shovel |  |
| Brute-Brainer Mk V Latrine Shovel |  |
| Orestes Mk XII Assault Chainaxe | content/items/weapons/player/melee/chainaxe_p1_m2 |
| Tigrus Mk XV Heavy Eviscerator | content/items/weapons/player/melee/chainsword_2h_p1_m2 |
| Cadia Mk XIII Assault Chainsword |  |
| Agripinaa Mk XIV Quickdraw Stub Revolver | content/items/weapons/player/ranged/stubrevolver_p1_m2 |
| Kantrael Mk X Heavy Laspistol | content/items/weapons/player/ranged/laspistol_p1_m3 |
| Catachan Mk VI Combat Blade | content/items/weapons/player/melee/combatknife_p1_m2 (?) |

**Comment by danreeves** [comment]
btw I’ll fix the extension when I get home on Sunday

**Comment by vtoadgames** [comment]
I was going to suggest that exception handling be added to allow the Chrome extension so that it displayed a placeholder if it can't match on a `display_name` but it looks like the Firefox extension already does this:

![image](https://github.com/danreeves/dt-exchange/assets/60324212/0befde3d-6dd1-4b17-bdaa-4ec8e8d465ec)

This is a good work-around for anyone who doesn't mind the seeing the `id` instead of the `display_name` until this can be updated.

**Comment by danreeves** [comment]
Yep, the chrome version would do that too but Google are blocking me updating until I change the extension to use a new manifest version
